src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -87,76 +89,34 @@ int32
 MultiClientConnect(const char *nodeName, uint32 nodePort, const char *nodeDatabase,
 				   const char *userName)
 {
-	PGconn *connection = NULL;
-	char connInfoString[STRING_BUFFER_SIZE];
+	MultiConnection *connection = NULL;
 	ConnStatusType connStatusType = CONNECTION_OK;
 	int32 connectionId = AllocateConnectionId();
-	char *effectiveDatabaseName = NULL;
-	char *effectiveUserName = NULL;
-
-	if (XactModificationLevel > XACT_MODIFICATION_NONE)
```

Comment:
I'm not particularly happy with that piece of logic, so I'm happy to move it around. But it's not generally clear how you can deal with it all the callsites, if you consider connection reusage making things safe.